### PR TITLE
Removing log dependency from ThreadingAndWaitlist test

### PR DESCRIPTION
**Pull request by saif-ericsson**
### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/test/java/com/ericsson/ei/testsuite/SuiteTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package com.ericsson.ei.testsuite;
```
Comment (a-breid-ericsson) [inline src/test/java/com/ericsson/ei/testsuite/SuiteTest.java]: Should this file be here?

File: src/test/java/com/ericsson/ei/testsuite/MongoPool.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.ericsson.ei.testsuite;
```
Comment (a-breid-ericsson) [inline src/test/java/com/ericsson/ei/testsuite/MongoPool.java]: Should this file be here?

File: src/test/java/com/ericsson/ei/subscription/SubscriptionHandlerTest.java [hunk 3]
```diff
@@ -65,6 +65,7 @@
         "rabbitmq.exchange.name: SubscriptionHandlerTest-exchange", "rabbitmq.consumerName: SubscriptionHandlerTest",
```
Comment (a-breid-ericsson) [inline src/test/java/com/ericsson/ei/subscription/SubscriptionHandlerTest.java]: Please revert changes in this file.

File: src/test/java/com/ericsson/ei/flowtests/TestConfigs.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
+package com.ericsson.ei.flowtests;
```
Comment (a-breid-ericsson) [inline src/test/java/com/ericsson/ei/flowtests/TestConfigs.java]: This file is used by many tests, not only in flow tests, please revert changes and move it back to test.utils

File: src/test/java/com/ericsson/ei/flowtests/FlowTest2.java [hunk 5]
```diff
@@ -42,6 +42,7 @@
 @TestPropertySource(properties = { "rules.path=src/test/resources/ArtifactRules.json",
```
Comment (a-breid-ericsson) [inline src/test/java/com/ericsson/ei/flowtests/FlowTest2.java]: Please revert changes back.

File: src/test/java/com/ericsson/ei/flowtests/FlowTest.java [hunk 6]
```diff
@@ -62,6 +62,7 @@
 @TestPropertySource(properties = { "rules.path=src/test/resources/ArtifactRules.json",
         "spring.data.mongodb.database: FlowTest", "rabbitmq.exchange.name: FlowTest-exchange",
         "rabbitmq.consumerName: FlowTestConsumer" })
+
```
Comment (a-breid-ericsson) [inline src/test/java/com/ericsson/ei/flowtests/FlowTest.java]: Please revert changes back.

File: src/functionaltests/resources/features/threadingAndWaitlistRepeat.feature [hunk 7]
```diff
@@ -6,6 +6,7 @@ Feature: Test Threading and Waitlist Repeat
   	Given that eiffel events are sent
 	Then waitlist should not be empty
 	And no event is aggregated
-	And the waitlist will try to resend the events at given time interval
+        And event-to-object-map is manipulated to include the sent events
```
Comment (a-breid-ericsson) [inline src/functionaltests/resources/features/threadingAndWaitlistRepeat.feature]: Please fix indention.

File: src/functionaltests/resources/ThreadingAndWaitlistRules.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (a-breid-ericsson) [inline src/functionaltests/resources/ThreadingAndWaitlistRules.json]: such small json data could be stored in a private static final variable and reduce the need to spend resources on reading from a file.

File: src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java [hunk 9]
```diff
@@ -80,45 +79,47 @@ public void no_event_is_aggregated() throws Throwable {
...
+        String dummyObjectID = "1234abcd-12ab-12ab-12ab-123456abcdef";
+        List<String> eventNames = getEventNamesToSend();
+        for (String eventName : eventNames) {
+            JsonNode eventJson = parsedJSON.get(eventName);
+            eventToObjectMapHanler.updateEventToObjectMapInMemoryDB(rulesObject, eventJson.toString(), dummyObjectID);
         }
-        assertEquals(getEventNamesToSend().size(), resentEvents.size());
+    }
+
+    @Then("^the next time waitlist will try to resend the events, they will get deleted$")
```
Comment (a-breid-ericsson) [inline src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java]: By the text given it one may think this test deletes them, better to have a name like "when waitlist has resent events they should have been deleted"

File: src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java [hunk 10]
```diff
@@ -4,40 +4,45 @@
...
+        "logging.level.com.ericsson.ei.handlers.EventHandler=OFF" })
 
 @Ignore
 public class ThreadingAndWaitlistRepeatSteps extends FunctionalTestBase {
-
-    private File tempLogFile;
     private static final String EIFFEL_EVENTS_JSON_PATH = "src/functionaltests/resources/eiffel_events_for_thread_testing.json";
+    private static final String INPUT_RULES_PATH = "src/functionaltests/resources/ThreadingAndWaitlistRules.json";
+
+    // private RulesObject rulesObject;
```
Comment (a-breid-ericsson) [inline src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java]: Remove commented if not needed

File: src/test/java/com/ericsson/ei/testsuite/SuiteTest.java (hunk 1 above)
Comment (saif-ericsson) [reply src/test/java/com/ericsson/ei/testsuite/SuiteTest.java]: the package testsuite is not needed and deleted

File: src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java (hunk 10 above)
Comment (saif-ericsson) [reply src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java]: done

File: src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java (hunk 9 above)
Comment (saif-ericsson) [reply src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java]: text changed

File: src/functionaltests/resources/ThreadingAndWaitlistRules.json (hunk 8 above)
Comment (saif-ericsson) [reply src/functionaltests/resources/ThreadingAndWaitlistRules.json]: It is changed as suggested

File: src/functionaltests/resources/features/threadingAndWaitlistRepeat.feature (hunk 7 above)
Comment (saif-ericsson) [reply src/functionaltests/resources/features/threadingAndWaitlistRepeat.feature]: done

File: src/test/java/com/ericsson/ei/flowtests/FlowTest.java (hunk 6 above)
Comment (saif-ericsson) [reply src/test/java/com/ericsson/ei/flowtests/FlowTest.java]: done

File: src/test/java/com/ericsson/ei/subscription/SubscriptionHandlerTest.java (hunk 3 above)
Comment (saif-ericsson) [reply src/test/java/com/ericsson/ei/subscription/SubscriptionHandlerTest.java]: done

File: src/test/java/com/ericsson/ei/flowtests/FlowTest2.java (hunk 5 above)
Comment (saif-ericsson) [reply src/test/java/com/ericsson/ei/flowtests/FlowTest2.java]: done

File: src/test/java/com/ericsson/ei/flowtests/FlowTest.java (hunk 6 above)
Comment (saif-ericsson) [reply src/test/java/com/ericsson/ei/flowtests/FlowTest.java]: done

File: src/test/java/com/ericsson/ei/flowtests/TestConfigs.java (hunk 4 above)
Comment (saif-ericsson) [reply src/test/java/com/ericsson/ei/flowtests/TestConfigs.java]: done

File: src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java [hunk 11]
```diff
@@ -19,25 +16,29 @@
...
-        "logging.level.com.ericsson.ei.waitlist=DEBUG", "logging.level.com.ericsson.ei.handlers.EventHandler=DEBUG" })
+        "rabbitmq.consumerName: ThreadingAndWaitlistRepeatStepsConsumer", "logging.level.com.ericsson.ei.waitlist=OFF",
+        "logging.level.com.ericsson.ei.handlers.EventHandler=OFF" })
 
 @Ignore
 public class ThreadingAndWaitlistRepeatSteps extends FunctionalTestBase {
-
-    private File tempLogFile;
     private static final String EIFFEL_EVENTS_JSON_PATH = "src/functionaltests/resources/eiffel_events_for_thread_testing.json";
+    private static final String idRule = "{" + "\"IdRule\": \"meta.id\"" + "}";
```
Comment (a-breid-ericsson) [inline src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java]: App private static finals should be un upper and snake case.

File: src/functionaltests/resources/ThreadingAndWaitlistRules.json (hunk 8 above)
Comment (a-breid-ericsson) [reply src/functionaltests/resources/ThreadingAndWaitlistRules.json]: Then this file could be removed :)

File: src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java (hunk 11 above)
Comment (saif-ericsson) [reply src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java]: done

File: src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java [hunk 12]
```diff
@@ -80,45 +75,46 @@ public void no_event_is_aggregated() throws Throwable {
...
+        for (String eventName : eventNames) {
+            JsonNode eventJson = parsedJSON.get(eventName);
+            eventToObjectMapHanler.updateEventToObjectMapInMemoryDB(rulesObject, eventJson.toString(), dummyObjectID);
         }
-        assertEquals(getEventNamesToSend().size(), resentEvents.size());
+    }
+
+    @Then("^when waitlist has resent events they should have been deleted$")
+    public void when_waitlist_has_resent_events_they_should_have_been_deleted() throws Throwable {
+        TimeUnit.SECONDS.sleep(2);
```
Comment (tobiasake) [inline src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java]: Why a sleep?
Isn't it better to poll DB until all events has been deleted in the test. And if it takes too long time, use a timeout that interrupt the test.

**Pull request merged**